pkg/combustion/registry.go

```diff
@@ -30,50 +28,48 @@ var haulerManifest string
 var registryScript string
 
 func configureRegistry(ctx *image.Context) ([]string, error) {
-	if IsEmbeddedArtifactRegistryEmpty(ctx.ImageDefinition.EmbeddedArtifactRegistry) {
+	if IsEmbeddedArtifactRegistryAndKubernetesManifestsEmpty(ctx) {
 		log.AuditComponentSkipped(registryComponentName)
 		return nil, nil
 	}
 
-	err := writeHaulerManifest(ctx)
-	if err != nil {
-		log.AuditComponentFailed(registryComponentName)
-		return nil, fmt.Errorf("writing hauler manifest: %w", err)
-	}
-
-	err = populateHaulerStore(ctx)
+	haulerBinaryPath := fmt.Sprintf("hauler-%s", string(ctx.ImageDefinition.Image.Arch))
+	err := copyHaulerBinary(ctx, haulerBinaryPath)
 	if err != nil {
 		log.AuditComponentFailed(registryComponentName)
-		return nil, fmt.Errorf("populating hauler store: %w", err)
+		return nil, fmt.Errorf("copying hauler binary: %w", err)
 	}
 
-	err = generateRegistryTar(ctx)
+	registryScriptNameResult, err := writeRegistryScript(ctx)
 	if err != nil {
 		log.AuditComponentFailed(registryComponentName)
-		return nil, fmt.Errorf("generating hauler store tar: %w", err)
+		return nil, fmt.Errorf("writing registry script: %w", err)
 	}
 
-	haulerBinaryPath := fmt.Sprintf("hauler-%s", string(ctx.ImageDefinition.Image.Arch))
-	err = copyHaulerBinary(ctx, haulerBinaryPath)
+	registriesDir := filepath.Join(ctx.CombustionDir, "registry")
+	err = os.Mkdir(registriesDir, os.ModePerm)
 	if err != nil {
 		log.AuditComponentFailed(registryComponentName)
-		return nil, fmt.Errorf("copying hauler binary: %w", err)
+		return nil, fmt.Errorf("creating registry dir: %w", err)
 	}
 
-	registryScriptNameResult, err := writeRegistryScript(ctx)
-	if err != nil {
-		log.AuditComponentFailed(registryComponentName)
-		return nil, fmt.Errorf("writing registry script: %w", err)
+	if len(ctx.ImageDefinition.EmbeddedArtifactRegistry.ContainerImages) != 0 {
+		err = writeHaulerManifestAndGenerateTar(ctx, ctx.ImageDefinition.EmbeddedArtifactRegistry.ContainerImages, nil, "embedded-artifact-registry")
+		if err != nil {
+			log.AuditComponentFailed(registryComponentName)
+			return nil, fmt.Errorf("writing hauler manifest and generating registry tar: %w", err)
+		}
 	}
 
 	log.AuditComponentSuccessful(registryComponentName)
 	return []string{registryScriptNameResult}, nil
 }
 
-func writeHaulerManifest(ctx *image.Context) error {
+func writeHaulerManifest(ctx *image.Context, images []image.ContainerImage, charts []image.HelmChart, haulerManifestYamlName string) error {
 	haulerManifestYamlFile := filepath.Join(ctx.BuildDir, haulerManifestYamlName)
+	registryDef := image.EmbeddedArtifactRegistry{ContainerImages: images, HelmCharts: charts}
 
-	data, err := template.Parse(haulerManifestYamlName, haulerManifest, ctx.ImageDefinition.EmbeddedArtifactRegistry)
+	data, err := template.Parse(haulerManifestYamlName, haulerManifest, registryDef)
```

Comment:
A unified manifest is not what we want if we're creating separate tarballs. This makes the logic too hard to follow as well as having empty manifest sections.